forge-game/src/main/java/forge/game/cost/CostCollectEvidence.java

```diff
@@ -0,0 +1,70 @@
+package forge.game.cost;
+
+import forge.game.Game;
+import forge.game.ability.AbilityKey;
+import forge.game.ability.SpellAbilityEffect;
+import forge.game.card.Card;
+import forge.game.card.CardCollection;
+import forge.game.card.CardLists;
+import forge.game.player.Player;
+import forge.game.spellability.SpellAbility;
+import forge.game.zone.ZoneType;
+
+import java.util.Map;
+
+public class CostCollectEvidence extends CostPartWithList {
+    // CollectEvidence<Amount>
+
+    public CostCollectEvidence(final String amount) {
+        this.setAmount(amount);
+    }
+
+    public static final String HashLKIListKey = "Collected";
+    public static final String HashCardListKey = "CollectedEvidence";
+
+    @Override
+    public String getHashForLKIList() {
+        return HashLKIListKey;
+    }
+    @Override
+    public String getHashForCardList() {
+        return HashCardListKey;
+    }
+
+    @Override
+    public int paymentOrder() { return 13; }
+
+    @Override
+    public boolean canPay(SpellAbility ability, Player payer, boolean effect) {
+        CardCollection list = new CardCollection(payer.getCardsIn(ZoneType.Graveyard));
```

Comment:
getTotalCMC should just accept this directly without creating a new object

Reply:
Ah yep. I was doing something more robust earlier until I found that helper function.